app/views/UserNotificationPreferencesView/ListPicker.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { StyleSheet, Text } from 'react-native';
+
+import * as List from '../../containers/List';
+import I18n from '../../i18n';
+import { useTheme } from '../../theme';
+import sharedStyles from '../Styles';
+import { OPTIONS } from './options';
+import { CustomIcon } from '../../containers/CustomIcon';
+import { useActionSheet } from '../../containers/ActionSheet';
+
+const styles = StyleSheet.create({
+	pickerText: {
+		...sharedStyles.textRegular,
+		fontSize: 16
+	}
+});
+
+type TKey = 'desktopNotifications' | 'pushNotifications' | 'emailNotificationMode';
+
+interface IBaseParams {
+	preference: TKey;
+	value: string;
+	onChangeValue: (param: { [key: string]: string }, onError: () => void) => void;
+}
+
+const ListPicker = ({
+	preference,
+	value,
+	title,
+	testID,
+	onChangeValue
+}: {
+	title: string;
+	testID: string;
+} & IBaseParams) => {
+	const { showActionSheet, hideActionSheet } = useActionSheet();
+	const { colors } = useTheme();
+	const [option, setOption] = useState(
+		value ? OPTIONS[preference].find(option => option.value === value) : OPTIONS[preference][0]
+	);
+
+	const getOptions = () =>
```

Comment:
Yeap, will work if will be only a constant, however, every time the Picker re-render, will trigger a map, but this map we only need when opening the action sheet. So, instead of do a map every re-render, we are trigger the map only when we try to open the showActionSheet